catalog_llm_query/app.py

```diff
@@ -0,0 +1,167 @@
+from flask import Flask, request, jsonify
+import json
+from arango import ArangoClient
+from langchain_community.graphs import ArangoGraph
+from langchain.chains import ArangoGraphQAChain
+from langchain_openai import ChatOpenAI
+import os
+from select_collections import select_collections
+from langchain_community.callbacks import get_openai_callback
+
+
+# Initialize Flask app
+app = Flask(__name__)
+
+
+def get_config():
+    with open('config/development.json') as f:
+        config = json.load(f)
+    return config
+
+
+def initialize_arango_graph(config):
+    # Configuration for ArangoDB
+    connection_uri = config['connectionUri']
+    dbName = config['dbName']
+    username = config['auth']['username']
+    password = config['auth']['password']
+
+    # Connect to ArangoDB and initialize graph
+    client = ArangoClient(hosts=connection_uri)
+    db = client.db(dbName, username=username, password=password)
+
+    # Initialize ArangoGraph
+    graph = ArangoGraph(db)
+    return graph
+
+
+def initialize_collection_names(collection_schema):
+    collection_names = [collection['collection_name']
+                        for collection in collection_schema]
+    return collection_names
+
+
+def initialize_llm(config):
+    api_key = config['openai_api_key']
+    os.environ['OPENAI_API_KEY'] = api_key
+    model = ChatOpenAI(temperature=0, model_name='gpt-4o')
+    return model
+
+
+def ask_llm(question):
+    selected_collection_names = select_collections(question, collection_names)
+    print('collection to use for query:', selected_collection_names)
```

Comment:
is this a log or an intended print for each query?

Reply:
will remove it.